parsl/executors/high_throughput/executor.py

```diff
@@ -609,14 +618,42 @@ def scale_in(self, blocks=None, block_ids=[]):
         if block_ids:
             block_ids_to_kill = block_ids
         else:
-            block_ids_to_kill = list(self.blocks.keys())[:blocks]
-
+            managers = self.connected_managers
+            blks = {}
+            for manager in managers:
+                if not manager['active']:
+                    continue
+                b_id = manager['block_id']
+                if b_id not in blks:
+                    blks[b_id] = [0, float('inf')]
+                blks[b_id][0] += manager['tasks']
+                blks[b_id][1] = min(blks[b_id][1], manager['idle_duration'])
+
+            sorted_blks = sorted(blks.items(), key=lambda item: (item[1][1], item[1][0]))
+            logger.warning("YADU: sorted_block : {}".format(sorted_blks))
+            if force is True:
+                block_ids_to_kill = [x[0] for x in sorted_blks[:blocks]]
+            else:
+                if not max_idletime:
+                    block_ids_to_kill = [x[0] for x in sorted_blks[:blocks] if x[1][0] == 0]
```

Comment:
I think this can select fewer than `blocks` blocks because it only examines the first `blocks` blocks to see if they are without tasks. (truncate first, then filter).

Maybe it should filter first, then truncate, which is what the `if max_idletime` clause below does.